internal/describer/tab.go

```diff
@@ -0,0 +1,137 @@
+/*
+ *  Copyright (c) 2020 the Octant contributors. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package describer
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/vmware-tanzu/octant/internal/link"
+	"github.com/vmware-tanzu/octant/internal/log"
+	"github.com/vmware-tanzu/octant/internal/modules/overview/logviewer"
+	"github.com/vmware-tanzu/octant/internal/modules/overview/terminalviewer"
+	"github.com/vmware-tanzu/octant/internal/modules/overview/yamlviewer"
+	"github.com/vmware-tanzu/octant/internal/printer"
+	"github.com/vmware-tanzu/octant/internal/resourceviewer"
+	"github.com/vmware-tanzu/octant/pkg/view/component"
+)
+
+func CustomResourceSummaryTab(crdName string) TabFactory {
```

Comment:
I'd prefer to see all of these share the same signature. I'd expect them all to return a TabFactory, not some of them also directly be a TabFactory based on the naming convention. My initial thought is to make them all return a TabFactory and create a TabFactoryOptions type they can accept for the factories that need setup before we do Component generation.